src/routes/Profile/BadgeCard/BadgeCard.tsx

```diff
@@ -27,53 +29,70 @@ export type BadgeCardProps = {
   disableClaiming: boolean;
 };
 
-export const BadgeCard: React.FunctionComponent<BadgeCardProps> = ({
-  onClaimButtonClick,
-  claimButtonMode,
-  loading,
-  variant,
-  claimedAt,
-  disableClaiming,
-}) => {
-  return (
-    <Container data-testid="BadgeCard">
-      <BadgePillBox>
-        <BadgePill loading={loading} variant={variant} />
-      </BadgePillBox>
-      <BadgeBox>
-        <Badge loading={loading} variant={variant} />
-      </BadgeBox>
-      <TitleBox>
-        {loading ? (
-          <TitleSkeleton variant="text" />
-        ) : (
-          <TitleTypography variant="body2">
-            {BADGE_VARIANT_TITLE_COPY_MAP[variant].toUpperCase()}
-          </TitleTypography>
-        )}
-      </TitleBox>
-      <DescriptionBox>
-        {loading ? (
-          <DescriptionSkeleton variant="text" />
-        ) : (
-          <DescriptionTypography variant="body2">
-            {BADGE_VARIANT_DESCRIPTION_COPY_MAP[variant]}
-          </DescriptionTypography>
-        )}
-      </DescriptionBox>
-      {disableClaiming ? null : (
-        <ClaimButtonBox>
+export type BadgeCardHandle = {
+  scrollIntoView: () => void;
+};
+
+export const BadgeCard = forwardRef<BadgeCardHandle, BadgeCardProps>(
+  ({ onClaimButtonClick, claimButtonMode, loading, variant, claimedAt, disableClaiming }, ref) => {
+    const containerRef = useRef<HTMLElement>(null);
+    const [highlight, setHighlight] = useState(false);
+    useImperativeHandle(ref, () => ({
```

Comment:
what's the use-case of this hook?

Reply:
useImperativeHandle is a hook from React. It allows you to have a component and expose an external API for other components to use.

For example: 

```
const Parent = () => {
   const ref = useRef();
   return (
     <>
        <Child ref={ref} />
        <button onClick={() => ref.current.sayHello()} />
     </>
   )
}

const Child = React.forwardRef((props, ref) => {
   useImperativeHandle(() => ({ sayHello: alert('hello') });
   return <div>Child</div>;
});
```

We define 2 components, Child and Parent. The Child exposes API for the parent to use in any way possible. Hope this makes sense. If not I can go a bit deeper.

In the current use case, I made BadgeCard to expose and API called highlight so that when someone wants to highlight the BadgeCard then can do that by calling ref.highlight();

Reply:
i see, so the parent can reference a given child in its logic via the reference, the imperative handle is necessary for the ref constant in the parent component to work, is that correct interpretation?